Create v1-only branch

sgallagher opened the pull request:
This branch will be maintenance-only for libmodulemd 1.x.

This patch drops all v2 code from the tree and reworks the
build system and CI tests for this branch.

Signed-off-by: Stephen Gallagher <sgallagh@redhat.com>

mmathesius commented on modulemd/meson.build:
```diff
@@ -34,244 +36,413 @@ valgrind = find_program('valgrind', required: developer_build)
+# Test env with fatal warnings and criticals
+test_env = test_release_env
+test_env.set('G_DEBUG', 'fatal-warnings,fatal-criticals')
+
```
General question... Is there any reason you chose to go with tediously itemizing the individual tests below, instead of looping through a map like you did for the v2 split? Or, did you simply pull in the tests from the former `v1/meson.build` file and strip the "_v1" just to be done with it? https://github.com/fedora-modularity/libmodulemd/blob/ee2b4615e75f388b5efdc29ab4609b2d65007bfe/modulemd/meson.build#L297-L313

sgallagher replied:
I just copied them in and stripped the _v1. This branch is not likely to see much activity; I just want it out of the v2 tarball.

sgallagher commented on modulemd/meson.build:
```diff
@@ -291,34 +462,66 @@ test('pycodestyle', pycodestyle,
 
-valgrind_tests = []
-
-if build_api_v1
```
No, it was moved elsewhere in the file.

mmathesius replied:
Sorry, but I just can't find a `test_import_headers` or anything that makes use of the `test-import-headers.sh` script any more.

sgallagher replied:
Hmm, I may have dropped it accidentally. I'll restore it. Thanks for catching that.

sgallagher commented:
> `contrib/doc-tools/fix-xref.sh` can also be dropped (or it could be put to use when generating the v1 documentation).

Yeah, dropped. We don't publish the 1.0 documentation to the web, so it's not needed.

> `README.md` also needs updating. The examples are for the 2.0 API, right? "Building the sources" section needs updated to reflect the dropping of the `build_api_v*` options, change the build directory name, and indicate that it's building the 1.0 API.

I dropped all of the examples. It's a deprecated API; I'm not willing to go through and write a good tutorial for it at this point.

mmathesius commented on libmodulemd.spec.in:
```diff
@@ -1,13 +1,5 @@
-%global libmodulemd_v1_version @V1_VERSION@
-
-%if ( 0%{?fedora} && 0%{?fedora} <= 30 ) || ( 0%{?rhel} && 0%{?rhel} <= 7)
-%global build_python2 1
```
The usage of `%if %{build_python2}`  below now needs to be dropped.

sgallagher replied:
That was a mistake. I only should have deleted the meson flags, not  `%build_python2`. I just pushed a fix.

mmathesius commented:
> I dropped all of the examples. It's a deprecated API; I'm not willing to go through and write a good tutorial for it at this point.

Works for me!

sgallagher commented:
I restored the header import test and squashed the patches together.
